skopt/utils.py

```diff
@@ -44,3 +52,58 @@ def create_result(Xi, yi, space=None, rng=None, specs=None, models=None):
     res.random_state = rng
     res.specs = specs
     return res
+
+
+def dump(res, filename):
+    """
+    Store an skopt optimization result into a file.
+
+    If the objective function (`.specs['args']['func']`) is unserializable,
+    it will be removed before saving into file.
+
+    Parameters
+    ----------
+    * `res` [`OptimizeResult`, scipy object]:
+        Optimization result object to be stored.
+
+    * `filename` [string]:
+        The path of the file in which it is to be stored.
+    """
+
+    try:
+        with open(filename, 'wb') as f:
+            pickle.dump(res, f)
```

Comment:
Somewhere we discussed whether or not to always use `joblib`. I think it was about "is it necessary to use `joblib` or would `pickle` suffice?" Can't find that thread right now. I'd be happy simply using `joblib` always.